src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -243,80 +337,36 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
-    if (unsortedUnavailableItems == null || unsortedUnavailableItems.isEmpty()) {
-      return CompletableFuture.completedFuture(succeeded(records));
-    }
+  private Result<InstanceRequestRelatedRecords> combineWithUnavailableItems(
+    Result<Map<Item, DateTime>> itemDueDateMapResult,
+    Result<Map<Item, Integer>> itemRequestQueueSizeMapResult,
+    InstanceRequestRelatedRecords records){
 
-    for (Item item : unsortedUnavailableItems) {
-      itemRequestQueueMap.put(item, queueRepository.getRequestQueueWithoutItemLookup(item.getItemId()));
-    }
+    return of(() -> {
 
-    final Collection<CompletableFuture<Result<RequestQueue>>> requestQueueFutures = itemRequestQueueMap.values();
+      if (itemDueDateMapResult.succeeded() && itemRequestQueueSizeMapResult.succeeded()) {
+        Map<Item, DateTime> itemDueDateMap = itemDueDateMapResult.value();
+        Map<Item, Integer> itemQueueSizeMap = itemRequestQueueSizeMapResult.value();
 
-    //Collect the RequestQueue objects once they come back
-    return CompletableFuture.allOf(requestQueueFutures.toArray(new CompletableFuture[requestQueueFutures.size()]))
-      .thenApply(x -> {
-        Map<Item, RequestQueue> itemQueueSizeMap = new HashMap<>();
-        List<Item> failedQueuesItemList = new LinkedList<>(); //to preserve the order it was found.
-
-        for (Map.Entry<Item, CompletableFuture<Result<RequestQueue>>> entry : itemRequestQueueMap.entrySet()) {
-          Result<RequestQueue> requestQueueResult = entry.getValue().join();
-          if (requestQueueResult.succeeded()) {
-            itemQueueSizeMap.put(entry.getKey(), requestQueueResult.value());
-          } else {
-            failedQueuesItemList.add(entry.getKey());
-          }
+        if (itemDueDateMap == null && itemQueueSizeMap == null) {
+          return records;
         }
 
-        if (failedQueuesItemList.size() == requestQueueFutures.size()
-            && (records.getSortedAvailableItems() == null || records.getSortedAvailableItems().isEmpty())) {
-          //fail the requests when there are no items to make requests from.
-          log.error("Failed to find request queues for all items of instanceId {}",
-                        unsortedUnavailableItems.get(0).getInstanceId());
-          return failed(new ServerErrorFailure("Unable to find an item to place a request"));
-        }
         //Sort the map
-        Map<Item, RequestQueue> sortedMap = sortRequestQueues(itemQueueSizeMap);
+        Map<Item, Integer> sortedMap = sortRequestQueues(itemQueueSizeMap, itemDueDateMap,
+          records.getInstanceLevelRequest().getPickupServicePointId());
 
-        LinkedList<Item> finalOrdedList = new LinkedList<>();
-
-        finalOrdedList.addAll(sortedMap.keySet());
-        //put the items that weren't able to retrieve RequestQueues for on the bottom of the list
-        finalOrdedList.addAll(failedQueuesItemList);
+        LinkedList<Item> finalOrdedList = new LinkedList<>(sortedMap.keySet());
```

Comment:
Just curious, why the `LinkedList` here instead of an `ArrayList`? There aren't too many situations where `LinkedList` is preferred.

Reply:
Here I used LinkedList because I didn't want the sortedMap to be unsorted when converting to a general list. After making the change to ArrayList the tests still pass so I will keep it as LinkedList.